daemon/logger/factory/factory.go

```diff
@@ -0,0 +1,57 @@
+package factory
+
+import (
+	"fmt"
+	"sync"
+
+	logger "github.com/docker/docker/daemon/logger"
+)
+
+// Creator is a method that builds a logging driver instance with given context
+type Creator func(LoggingContext) (logger.Logger, error)
+
+// LoggingContext provides enough information for a logging driver to do its function
+type LoggingContext struct {
```

Comment:
Method inapplicable to all drivers now. And won't be applicable for one more in a week.
I think daemon should be aware about json-file logger, because it was designed without idea of drivers in mind. "path" field not forcing other developers to write useless methods for their drivers btw, so, ya, it is better.